prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/migrationtoolkit/GetAnalysisTask.java

```diff
@@ -0,0 +1,120 @@
+	// created using the context params serverURL and bearerToken during execution time.
+	// Not depending on the context params for the client details
+	// means flow authors can build the client object with their
+	// params and pass it is. This also prevents an invoker to pass the client
+	// details by ignoring them if a client already exists.
+	// This method is useful for testing as well.
+	protected MTATaskGroupClient mtaClient;
+
+	private URI serverUrl;
+
+	private final Consumer<Message> messageConsumer;
+
+	public GetAnalysisTask(URI serverURL, String bearerToken, Consumer<Message> messageConsumer) {
+		this.serverUrl = serverURL;
+		this.mtaClient = new MTAClient(serverURL, bearerToken);
+		this.messageConsumer = messageConsumer;
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key("taskGroupID").type(WorkParameterType.NUMBER).optional(false)
+						.description("The application name as presented in the application hub").build(),
+				WorkParameter.builder().key("serverURL").type(WorkParameterType.TEXT).optional(true).description(
+						"Base URL of the MTA instance - e.g https://mta-openshift-mta.app.clustername.clusterdomain")
+						.build(),
+				WorkParameter.builder().key("bearerToken").type(WorkParameterType.TEXT).optional(true)
+						.description("Bearer token to authenticate server requests").build());
+	}
+
+	/**
+	 * @param workContext optional context values: serverURL, and bearerToken for the
+	 * mtaClient.
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		if (mtaClient == null) {
+			serverUrl = URI.create(getOptionalParameterValue(workContext, "serverURL", null));
+			var bearerToken = getOptionalParameterValue(workContext, "bearerToken", null);
+			if (serverUrl == null) {
+				log.error(
+						"serverURL is empty. Either pass it while creating the instance of the task or in the context");
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext);
+			}
+			mtaClient = new MTAClient(serverUrl, bearerToken);
+		}
+
+		int taskGroupID;
+		try {
+			taskGroupID = Integer.parseInt(getRequiredParameterValue(workContext, "taskGroupID"));
+		}
+		catch (MissingParameterException | NumberFormatException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		Result<TaskGroup> result = mtaClient.get(taskGroupID);
+
+		if (result == null) {
+			// unexpected
+			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(),
+					new IllegalStateException("MTA client returned empty result with no error."));
+		}
+		else if (result instanceof Result.Failure<TaskGroup> failure) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, failure.t());
+		}
+		else if (result instanceof Result.Success<TaskGroup> success) {
+			if ("Ready".equals(success.value().state()) && success.value().tasks() != null
```

Comment:
if `success.value().tasks()` is not null, is it guaranteed it won't be empty as well?
since next, there is an access to the first element.